PatchNotes/Altibase_7.1/eng/Altibase_7_1_0_9_3_Patch_Notes.md

```diff
@@ -0,0 +1,585 @@
+      }
+      ```
+
+-   **Workaround**
+
+-   **Changes**
+
+    -   Performance view
+    -   Property
+    -   Compile Option
+    -   Error Code
+
+### BUG-50778<a name=bug-50778></a> Add exception handling to the getColumnCount function.
+
+-   **module** : qp
+-   **Category** : Fatal
+-   **Reproducibility** : Rare
+-   **Description** : Altibase internally invokes the getColumnCount function to request column information during the Prepare process. There was an issue that the server was abnormally terminated because of incorrect data structure access in an exceptional situation. Altibase added exception handling to this situation.
+-   **How to reproduce this bug**
+    -   **Reproduction conditions**
+    -   **Actual Results**
+    -   **Expected Results**
+
+-   **Workaround**
+-   **Changes**
+
+    -   Performance view
+    -   Property
+    -   Compile Option
+    -   Error Code
+
+### BUG-50783<a name=bug-50783></a> Resolve an issue where an "Invalid use of host variables" error occurred during DDL on a table while executing a procedure.
+
+-   **module** : qp
+
+-   **Category** : Functional Error
+
+-   **Reproducibility** : Always
+
+-   **Description** : Altibase resolved an issue where an  "ERR-3123B : Invalid use of host variables" error occurred during DDL on a table while executing a procedure.
+    
+- **How to reproduce this bug**
+
+  - **Reproduction conditions**
+
+    ```sql
+    -- session 1
+    CREATE TABLE T1 (I1 INTEGER);
+    INSERT INTO T1 VALUES(1);
+    CREATE OR REPLACE PROCEDURE PROC1 AS
+    VAR1 INTEGER;
+    VAR2 INTEGER;
+    BEGIN
+      VAR1 := 1;
+      FOR I IN 1 .. 10 LOOP
+      SELECT VAR1 INTO VAR2 FROM T1;
+      PRINTLN(VAR2);
+      SLEEP(2);
+      VAR1 := VAR1 + 1;
+      END LOOP;
+    END;
+    /
+    EXEC PROC1;
+    
+    -- session 2
+    ALTER TABLE T1 ADD (I2 INTEGER);
+    ```
+
+  - **Actual Results**
+
+    ```sql
+    iSQL> EXEC PROC1;
+    1
+    2
+    3
+    4
+    [ERR-3123B : Invalid use of host variables
+    0001 : select :B0 from T1;
+                 ^  ^
+    
+    at "SYS.PROC1", line 7]
+    ```
+
+  -   **Expected Results**
+
+      ```sql
+      iSQL> EXEC PROC1;
+      1
+      2
+      3
+      4
+      5
+      6
+      7
+      8
+      9
+      10
+      Execute success.
+      ```
+
+-   **Workaround**
+
+-   **Changes**
+
+    -   Performance view
+    -   Property
+    -   Compile Option
+    -   Error Code
+
+### BUG-50789<a name=bug-50789></a> Fix a problem where media recovery failed when additional checkpoint image files were added during the process.
+
+-   **module** : sm
+-   **Category** : Fatal
+-   **Reproducibility** : Always
+-   **Description** : When additional checkpoint image files were added during the MEDIA RECOVERY process, the media recovery failed because it was not able to find LSNs of the added checkpoint image files. To resolve this, now Altibase records CreateLSN of the added file during the MEDIA RECOVERY process on its header.
```

Comment:
```suggestion
-   **Description** : When additional checkpoint image files were added during the MEDIA RECOVERY process, the media recovery failed because it could not find LSNs of the added checkpoint image files. To resolve this, now Altibase records CreateLSN on the file header that are added during the MEDIA RECOVERY process.
```